### ISSUE TITLE
ag Crash Report

### ISSUE THREAD

**Issue by Awea** [rcoh/angle-grinder#138]
Hello there!

I'm trying ag to explore some logs, thanks for this tool <3. It had a problem and crashed so here I am reporting it.

Generated report file content:

```toml
name = 'ag'
operating_system = 'unix:Unknown'
crate_version = '0.16.0'
explanation = '''
Panic occurred in file '/rustc/d3fb005a39e62501b8b0b356166e515ae24e2e54/src/libcore/str/mod.rs' at line 1920
'''
cause = '''byte index 24 is not a char boundary; it is inside 'é' (bytes 23..25) of `"Bourgogne-Franche-Comté"'''
method = 'Panic'
backtrace = '''

   0: 0x56419e69e93a - core::option::Option<T>::unwrap_or_else::hc7a102dd836873c2
                at /rustc/d3fb005a39e62501b8b0b356166e515ae24e2e54/src/libcore/option.rs:430
                 - core::str::traits::<impl core::slice::SliceIndex<str> for core::ops::range::Range<usize>>::index::hbda1af65bab2f710
                at /rustc/d3fb005a39e62501b8b0b356166e515ae24e2e54/src/libcore/str/mod.rs:1920
                 - core::str::traits::<impl core::ops::index::Index<I> for str>::index::h3c3e06bc876f6edd
                at /rustc/d3fb005a39e62501b8b0b356166e515ae24e2e54/src/libcore/str/mod.rs:1780
                 - ag::operator::split::find_close_delimiter::h8be72b90f49ea10e
                at /home/awea/.asdf/installs/rust/1.45.2/registry/src/github.com-1ecc6299db9ec823/ag-0.16.0/src/operator/split.rs:25
                 - ag::operator::split::split_with_delimiters::h6fa224ad1753a2f3
                at /home/awea/.asdf/installs/rust/1.45.2/registry/src/github.com-1ecc6299db9ec823/ag-0.16.0/src/operator/split.rs:64
   1: 0x56419e68dbd3 - <ag::operator::Split as ag::operator::UnaryPreAggFunction>::process::h1bd51bb0e53cbea0
                at /home/awea/.asdf/installs/rust/1.45.2/registry/src/github.com-1ecc6299db9ec823/ag-0.16.0/src/operator.rs:1011
   2: 0x56419e50132c - <T as ag::operator::UnaryPreAggOperator>::process_mut::hc22aff7648e86bd9
                at /home/awea/.asdf/installs/rust/1.45.2/registry/src/github.com-1ecc6299db9ec823/ag-0.16.0/src/operator.rs:146
   3: 0x56419e65f2de - ag::pipeline::Pipeline::proc_preagg::h24c2cb2a3184524e
                at /home/awea/.asdf/installs/rust/1.45.2/registry/src/github.com-1ecc6299db9ec823/ag-0.16.0/src/lib.rs:323
   4: 0x56419e50bc63 - ag::pipeline::Pipeline::process::h752affbf09a5f439
                at /home/awea/.asdf/installs/rust/1.45.2/registry/src/github.com-1ecc6299db9ec823/ag-0.16.0/src/lib.rs:289
   5: 0x56419e514c32 - agrind::main::h1cdeef69545a5007
                at /home/awea/.asdf/installs/rust/1.45.2/registry/src/github.com-1ecc6299db9ec823/ag-0.16.0/src/bin/agrind.rs:126
   6: 0x56419e515f67 - std::rt::lang_start::{{closure}}::h1cb8cce70727cb5d
                at /rustc/d3fb005a39e62501b8b0b356166e515ae24e2e54/src/libstd/rt.rs:67
   7: 0x56419e79d588 - std::rt::lang_start_internal::{{closure}}::hefb489d56a389b75
                at /rustc/d3fb005a39e62501b8b0b356166e515ae24e2e54/src/libstd/rt.rs:52
                 - std::panicking::try::do_call::hb2bfa58b743656f1
                at /rustc/d3fb005a39e62501b8b0b356166e515ae24e2e54/src/libstd/panicking.rs:297
                 - std::panicking::try::haaa85a89ce3784c3
                at /rustc/d3fb005a39e62501b8b0b356166e515ae24e2e54/src/libstd/panicking.rs:274
                 - std::panic::catch_unwind::h49baa277f74e60cb
                at /rustc/d3fb005a39e62501b8b0b356166e515ae24e2e54/src/libstd/panic.rs:394
                 - std::rt::lang_start_internal::h7aa2f6739b31fe58
                at /rustc/d3fb005a39e62501b8b0b356166e515ae24e2e54/src/libstd/rt.rs:51
   8: 0x56419e515b28 - main
   9: 0x7fe93e9720b3 - __libc_start_main
  10: 0x56419e4f63de - _start
  11:        0x0 - <unresolved>'''
```

**Comment by rcoh** [comment]
thanks! Looks like this line is wrong: https://github.com/rcoh/angle-grinder/blob/main/src/operator/split.rs#L25
it needs to account for the width of the codepoint

**Comment by rcoh** [comment]
Fix is here: #139 — not sure the next time I'll cut a release so if you need this fixed, I'd build from source.
